File: Leonardo Ogata/20/validParentheses.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        
        for(char c : s.toCharArray()) {
            if (c == '(' || c == '{' || c == '[') {
                stack.push(c);
            }

            if (c == ')' && stack.isEmpty()|| c == '}' && stack.isEmpty() || c == ']' && stack.isEmpty()) {
                return false;
            }

            if (c == ')') {
                char check = stack.peek(); 
                    if (check == '(') {
                        stack.pop();
                    }else{
                        return false;
                    }
            }

            if (c == '}') {
                char check = stack.peek(); 
                    if (check == '{') {
                        stack.pop();
                    }else{
                        return false;
                    }
            }

            if (c == ']') {
                char check = stack.peek(); 
                    if (check == '[') {
                        stack.pop();
                    }else{
                        return false;
                    }
            }
        }

        if (stack.isEmpty()) {
            return true;
        }else{
            return false;
        }
        
    
    }
}

File: giovanna-britto/20/Valid_Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let pilha = [];
    let abertoFechado = {
        ")": "(",
        "]": "[",
        "}": "{"
    };

    for (let i = 0; i < s.length; i++){
        if ((s[i] === "(") || (s[i] === "[") || (s[i] === "{")){
            pilha.push(s[i]);
        } else {
            if (pilha.pop() !== abertoFechado[s[i]]){
                return false;
            }
        }
    }

    return (pilha.length == 0);
};